packages/core-data/src/actions.js

```diff
@@ -944,25 +944,29 @@ export function receiveDefaultTemplateId( query, templateId ) {
  * @param {?Object}       query           Query Object.
  * @param {?boolean}      invalidateCache Should invalidate query caches.
  * @param {?Object}       meta            Meta information about pagination.
- * @return {Object} Action object.
  */
-export function receiveRevisions(
-	kind,
-	name,
-	recordKey,
-	records,
-	query,
-	invalidateCache = false,
-	meta
-) {
-	return {
-		type: 'RECEIVE_ITEM_REVISIONS',
-		items: Array.isArray( records ) ? records : [ records ],
-		recordKey,
-		meta,
-		query,
-		kind,
-		name,
-		invalidateCache,
+
+export const receiveRevisions =
+	( kind, name, recordKey, records, query, invalidateCache = false, meta ) =>
+	async ( { dispatch } ) => {
+		const configs = await dispatch( getOrLoadEntitiesConfig( kind ) );
+		const entityConfig = configs.find(
+			( config ) => config.kind === kind && config.name === name
+		);
+		const key =
```

Comment:
For me this should continue to use the default entity key.

Reply:
Do you mean `DEFAULT_ENTITY_KEY`? 

The value is `id`, which doesn't work for template/template parts. To get the unique record id we have to refer to the `wp_id` property.